src/gf.c

```diff
@@ -1379,8 +1381,10 @@ JL_DLLEXPORT int jl_compile_hint(jl_tupletype_t *types)
 
 JL_DLLEXPORT jl_value_t *jl_get_spec_lambda(jl_tupletype_t *types)
 {
-    jl_value_t *li = (jl_value_t*)jl_get_specialization1(types);
-    return li ? li : jl_nothing;
+    jl_lambda_info_t *li = jl_get_specialization1(types);
+    if (li && jl_generating_output() && jl_options.incremental && li->def->isstaged)
+        return jl_nothing; // TODO: precompile can't deal with this right now
```

Comment:
I don't understand why this restriction is needed. The relevant abstraction seems to be that precompile describes a LambdaInfo as "the LambdaInfo for this function for these types". Why does it matter whether a staged function was involved in generating the LambdaInfo?